src/main/scala/org/kframework/kore/implementation/Kore.scala

```diff
@@ -0,0 +1,140 @@
+package org.kframework.kore.implementation
+
+import org.kframework.kore.interfaces.{outer => o}
+import org.kframework.kore.interfaces.{pattern => p}
+import org.kframework.kore.interfaces.builders.OuterBuilders
+import org.kframework.kore.interfaces.builders.PatternBuilders
+
+import scala.collection._
+
+/** Algebraic data type of MiniKore. */
+object DefaultPattern {
+
+  /** A collection of classes that serve as the default implementation of the [[org.kframework.kore.interfaces.pattern]] **/
+
+  // Pattern Implementations
+  // =======================
+
+  case class Variable(_1: p.Name, _2: p.Sort) extends p.Variable {
+    def build(_1: p.Name, _2: p.Sort): Variable = Variable(_1, _2)
+  }
+
+  case class DomainValue(_1: p.Symbol, _2: p.Value) extends p.DomainValue {
+    def build(_1: p.Symbol, _2: p.Value): DomainValue = DomainValue(_1, _2)
+  }
+
+  case class Application(_1: p.Symbol, args: Seq[p.Pattern]) extends p.Application {
+    def build(_1: p.Symbol, args: Seq[p.Pattern]): Application = Application(_1, args)
+  }
+
+  case class Top() extends p.Top {
+    def build(): Top = Top()
+  }
+
+  case class Bottom() extends p.Bottom {
+    def build(): Bottom = Bottom()
+  }
+
+  case class And(_1: p.Pattern, _2: p.Pattern) extends p.And {
+    def build(_1: p.Pattern, _2: p.Pattern): And = And(_1, _2)
+  }
+
+  case class Or(_1: p.Pattern, _2: p.Pattern) extends p.Or {
+    def build(_1: p.Pattern, _2: p.Pattern): Or = Or(_1, _2)
+  }
+
+  case class Not(_1: p.Pattern) extends p.Not {
+    def build(_1: p.Pattern): Not = Not(_1)
+  }
+
+  case class Implies(_1: p.Pattern, _2: p.Pattern) extends p.Implies {
+    def build(_1: p.Pattern, _2: p.Pattern): Implies = Implies(_1, _2)
+  }
+
+  case class Exists(_1: p.Variable, _2: p.Pattern) extends p.Exists {
+    def build(_1: p.Variable, _2: p.Pattern): Exists = Exists(_1, _2)
+  }
+
+  case class ForAll(_1: p.Variable, _2: p.Pattern) extends p.ForAll {
+    def build(_1: p.Variable, _2: p.Pattern): ForAll = ForAll(_1, _2)
+  }
+
+  case class Next(_1: p.Pattern) extends p.Next {
+    def build(_1: p.Pattern): Next = Next(_1)
+  }
+
+  case class Rewrite(_1: p.Pattern, _2: p.Pattern) extends p.Rewrite {
+    def build(_1: p.Pattern, _2: p.Pattern): Rewrite = Rewrite(_1, _2)
+  }
+
+  case class Equals(_1: p.Pattern, _2: p.Pattern) extends p.Equals {
+    def build(_1: p.Pattern, _2: p.Pattern): Equals = Equals(_1, _2)
+  }
+
+}
+
+
+object DefaultOuter {
+
```

Comment:
@ehildenb I meant your first case.  As I said, from the perspective of one who wants to use K programmatically from Scala, the KORE API should be in a one-to-one correspondence to the KORE language.  Anything else will confuse people.  Even the very terminology, `Outer`, is confusing and adhoc.  Why `Outer`?  Radu came up with it at some moment in one of his codes, and somehow we ended up stuck with it.  I do not remember @cos' reasons for not merging the Outer and Pattern traits; can you recall them?  At this moment I think we should really separate the KORE interface into Main and Extended, as I said.  At least I can then push (hard!) for the Main interface to be faithful (identical, conceptually) to the KORE language.